### PR TITLE
Add a number of CSSOM interfaces + fix HTMLStyleElement.sheet type

### LINKED ISSUE #2943: Missing CSSStyleRule which extends CSSRule
Initially I thought `CSSRule` declaration should have `style` property, but no actually we miss an entire class `CSSStyleRule` which inherits from `CSSRule` and implements `style` property.

https://github.com/facebook/flow/blob/master/lib/cssom.js#L42

### PR DESCRIPTION
**Pull request by kof**
Fixes #2943 and #2696
- CSSStyleSheet - https://www.w3.org/TR/cssom-1/#the-cssstylesheet-interface
- CSSStyleRule - https://www.w3.org/TR/cssom-1/#cssstylerule
- CSSMediaRule - https://www.w3.org/TR/cssom-1/#the-cssmediarule-interface
- CSSConditionRule - https://developer.mozilla.org/en-US/docs/Web/API/CSSConditionRule
- CSSGroupingRule - https://www.w3.org/TR/2016/WD-cssom-1-20160317/#the-cssgroupingrule-interface

### CODE REVIEW

File: lib/cssom.js [hunk 1]
```diff
@@ -35,7 +35,26 @@ declare class CSSStyleSheet extends StyleSheet {
...
+  deleteRule(index: number): void;
+  insertRule(rule: string, index: number): number;
+}
+
+declare class CSSConditionRule extends CSSGroupingRule {
+  conditionText: string;
+}
+
+declare class CSSMediaRule extends CSSConditionRule {
+  media: MediaList;
```
Comment (mvitousek) [inline lib/cssom.js]: It looks like this attribute should be read-only, right? https://www.w3.org/TR/cssom-1/#the-cssmediarule-interface

Comment (mvitousek) [inline lib/cssom.js]: This should be read-only, right? https://www.w3.org/TR/2016/WD-cssom-1-20160317/#the-cssmediarule-interface

Comment (kof) [reply lib/cssom.js]: yep, I just went through all added interfaces and some previously existing once and updated all readonly props

**Comment by kof** [comment]
I wish I knew what is failing in Facebook internal builds.

**Comment by kof** [comment]
🎉

**Comment by mvitousek** [comment]
Thanks for the contribution, and sorry about the delay in merging!

**Pull request closed without merging**